backend/src/services/google_places_enhancer.py

```
@@ -250,4 +280,93 @@ async def enhance_attractions_batch(self, attractions: List[Dict], destination:
+                    if geocode_result:
+                        location = geocode_result[0]['geometry']['location']
+                        waypoints.append(f"{location['lat']},{location['lng']}")
+            
+            if not waypoints:
+                return {"error": "Could not get coordinates for attractions"}
+            
+            # Create route URL
+            if start_location:
+                route_url = f"https://maps.google.com/maps/dir/{start_location}/"
+            else:
+                route_url = f"https://maps.google.com/maps/dir/"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)